libass/ass.c

```diff
@@ -485,24 +498,20 @@ static int process_style(ASS_Track *track, char *str)
         // no style format header
         // probably an ancient script version
         if (track->track_type == TRACK_TYPE_SSA)
-            track->style_format = strdup(ssa_style_format);
+            track->style_format = strdup(ssa_style_format.str);
         else
-            track->style_format = strdup(ass_style_format);
+            track->style_format = strdup(ass_style_format.str);
         if (!track->style_format)
             return -1;
     }
 
-    q = format = strdup(track->style_format);
-    if (!q)
-        return -1;
+    ASS_StringView q = { track->style_format, strlen(track->style_format) }; // format scanning pointer
 
-    ass_msg(track->library, MSGL_V, "[%p] Style: %s", track, str);
+    ass_msg(track->library, MSGL_V, "[%p] Style: %*s", track, (int)p.len, p.str);
```

Comment:
Format specifier should be `%.*s` as string length is precision.

Reply:
Oh, whoops, good catch.